### PR TITLE
Setup Lock Screen

### PR DESCRIPTION
**Pull request by nZac**

### CODE REVIEW

File: keg_login/middleware.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    session_key_name = 'keg-login.lockout'
+
+    default_excluded_routes = [
+        'static',
+        'keg_login.logout',
+        lock_endpoint,
+    ]
+
+    def __init__(self, exclude=[]):
+        self.exclude_routes = exclude + self.default_excluded_routes
```
Comment (bladams) [inline keg_login/middleware.py]: NIT: the exclude routes could be sets/frozensets and even though it doesn't matter here, I think it's best to not use mutable default args

**Review by bladams — CHANGES_REQUESTED** [review]
Looks good to me. Just a couple minor nitpicks

File: keg_login/middleware.py (hunk 1 above)
Comment (rsyring) [reply keg_login/middleware.py]: It's more than a NIT IMO.  You should never have a mutable default argument like this.  In this case, the way it's used doesn't matter.  But I don't even want the possibility of that coming back to bite us later.

Comment (nZac) [reply keg_login/middleware.py]: Yea, that wasn't intentional. I just missed it while I was coding. Thanks @bladams.

Comment (nZac) [reply keg_login/middleware.py]: and @rsyring :)

File: keg_login/forms.py [hunk 2]
```diff
@@ -90,6 +90,15 @@ class ResetPasswordForm(Form):
     return ResetPasswordForm
 
 
+def make_lock_form():
```
Comment (rsyring) [inline keg_login/forms.py:93]: Why do we return the form from a function?

File: keg_login/middleware.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        self.exclude_routes = exclude | self.default_excluded_routes
+
+    def should_lock_out(self, endpoint, session):
+        if not session.get(self.session_key_name, False):
+            return False
+
+        return False if endpoint in self.exclude_routes else True
+
+    def __call__(self):
+        return (flask.redirect(flask.url_for(self.lock_endpoint))
```
Comment (rsyring) [inline keg_login/middleware.py]: I don't understand why this was written this way.  The straight forward form is easier to read and shorter:

```python
if self.should_lock_out(flask.request.endpoint, flask.session):
    return flask.redirect(flask.url_for(self.lock_endpoint))
```

File: keg_login/tests/test_middleware.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import keg_login.middleware as middleware
+
+
+class TestLockedOutMiddleware(object):
+    def test_excluded_routes(self):
+        cls = middleware.LockedOutMiddleware
```
Comment (rsyring) [inline keg_login/tests/test_middleware.py]: NIT: I'd avoid using `cls` as an assigned variable name in the same way I'd avoid using `self`.

**Review by rsyring — APPROVED** [review]
There are some things I'd do differently but I'm not opposed to shipping this.  Please answer the comments if you can.

File: keg_login/forms.py (hunk 2 above)
Comment (bladams) [reply keg_login/forms.py:93]: That seems to be the pattern for the rest of the forms in keg-login. I think this was just done for consistency.

**Review by bladams — APPROVED** [review]

**Pull request merged**